libhoney.go

```diff
@@ -146,6 +147,24 @@ type Config struct {
 	Logger Logger
 }
 
+func (c *Config) getDataset() string {
+	if c.isClassic() {
+		if strings.TrimSpace(c.Dataset) == "" {
+			return defaultClassicDataset
+		}
+		return c.Dataset
+	}
+	trimmedDataset := strings.TrimSpace(c.Dataset)
+	if trimmedDataset == "" {
+		return defaultDataset
+	}
+	return trimmedDataset
```

Comment:
warn on diff if we trimmed something (like we did in the [go beeline](https://github.com/honeycombio/beeline-go/blob/dcd5cac17320d77d7242d157e4dea6c757edbd03/beeline.go#L146))